src/modules/EventModule.cpp

```diff
@@ -0,0 +1,792 @@
+		double totalFee = (static_cast<double>(config.raffleDrawTicketPrice) * ticket);
+		CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+		auto& authorCurrency = currencyModule->guildList[channel.guildId()][currencyModule->getUserIndex(channel.guildId(), authorID)];
+
+		if (authorCurrency.currency() <= 0)
+		{
+			client.createMessage(message.channelId(), "**Haha you poor bruh. Git gud enough to afford tickets.**");
+			return;
+		}
+		if (totalFee > authorCurrency.currency())
+		{
+			client.createMessage(message.channelId(), "**Can you afford that much ticket?**");
+			return;
+		}
+		for (int i = 0; i != ticket; i++)
+		{
+			config.currentTicketIndex++;
+			configRD.m_TicketIds.push_back(config.currentTicketIndex);
+		}
+
+		authorCurrency.setCurrency(authorCurrency.currency() - totalFee);
+		QString ticketEmoji = QString(utility::consts::emojis::TICKETS);
+		QString ticketAmount = QString::number(configRD.m_TicketIds.size());
+		client.createMessage(message.channelId(), ticketEmoji + " **You own** `" + ticketAmount + "` **ticket(s) now!**\n *(Buy them more to increase your chance)*");
+	});
+	RegisterCommand(Commands::EVENT_TICKET, "ticket", [this](Client& client, const Message& message, const Channel& channel)
+	{
+		QStringList args = message.content().split(' ');
+		if (args.size() != 1)
+		{
+			client.createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+			return;
+		}
+		auto& configRD = raffleDrawGuildList[channel.guildId()][raffleDrawGetUserIndex(channel.guildId(), message.author().id())];
+		auto& config = getServerEventData(channel.guildId());
+		if (configRD.m_TicketIds.size() == 0)
+		{
+			client.createMessage(message.channelId(), "**You have no tickets.**\nYou can buy them by using `!buyticket <amount>`");
+			return;
+		}
+		QString name = UmikoBot::Instance().GetName(channel.guildId(), message.author().id());
+		QString totalTicket = QString::number(configRD.m_TicketIds.size());
+		QString ticketEmoji = QString(utility::consts::emojis::TICKETS);
+		QString ticketAmount = QString::number(configRD.m_TicketIds.size());
+		QString txt = QString(ticketEmoji + "Total Ticket(s) belonging to **%1**: `%2`\nYour ticket number(s) are:\n").arg(name, totalTicket);
+
+		for (auto ticket : configRD.m_TicketIds)
+		{
+			if (ticket == configRD.m_TicketIds.last())
+			{
+				txt += QString("%1").arg(ticket);
+				client.createMessage(message.channelId(), txt);
+				return;
+			}
+			txt += QString("%1, ").arg(ticket);
+		}
+	});
+	RegisterCommand(Commands::EVENT_SET_TICKET_PRICE, "setticketprice", [this](Client& client, const Message& message, const Channel& channel)
+	{
+		QStringList args = message.content().split(' ');
+
+		UmikoBot::VerifyAndRunAdminCmd(client, message, channel, 2, args, true, [this, &client, channel, message, args]()
+			{
+				auto& config = getServerEventData(channel.guildId());
+				config.raffleDrawTicketPrice = args.at(1).toInt();
```

Comment:
Most of the currency system currently uses `double`s (though we have plans to change it I guess), so one would assume that this also accepts `doubles`.